docs/DeveloperGuide.adoc

```diff
@@ -295,71 +316,174 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 [width="59%",cols="22%,<23%,<25%,<30%",options="header",]
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
-|`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
+|`* * *` |student from university courses|search problems by partially matching keywords|identify the problem I am required to do asap
 
-|`* * *` |user |add a new person |
+|`* * *` |developer|introduce new problems to the software easily |the repository of code challenges can be updated without much hassle
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* * *` |coding student|look back on the coding challenges I have completed|revise the concepts used in those problems
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |job seeker|view past interview problems by a company|increase my chances of getting hired
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * *` |user|see the difficulties of each problem|choose to do problems that are more aligned to my standard
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
-|=======================================================================
+|`* *` |developer|receive detailed auto-generated error reports if any bug occurs|correct them
+
+|`* *` |forgetful user|set reminders|be reminded of the problems that I need to solve before a deadline
+
+|`* *` |unorganized coder|view my progress on different categories|know which area I am weak in
+
+|`* *` |programming course student |attempt problems under timed condition|I feel more prepared in timed assessments such as labs, practical exam and final exam.
+
+|`* *` |tutor|choose to reveal the answers to the solutions through a password|
+
+|`* *` |coder|identify the concepts required to solve a problem before attempting them |move on to another quickly
+
+|`* *` |a coding student|attempt the same problem in different coding languages|test my proficiency at those languages
+
+|`* *` |coding amateur|look at hints/tutorials for the problem|learn something new while attempting a coding challenge
```

Comment:
Just a minor change. Coding student and coding amateur is not consistent, should put it as "student coder" and "amateur coder"